### PR TITLE
WIP: tabbook multi weights

### PR DESCRIPTION
**Pull request by 1beb**

### CODE REVIEW

**Comment by 1beb** [comment]
Please ignore. Just PR for CI.

File: R/tab-book.R [hunk 1]
```diff
@@ -46,11 +51,149 @@
...
+    # Add an index value per weight (or non-weight)
+    # This is used in repacking
+    tabFrame$index <- NA
+    for (w in unique(tabFrame$ind)) {
+        tabFrame$index[tabFrame$ind %in% w] <- seq_len(
+            nrow(tabFrame[tabFrame$ind %in% w,])
+        )
+    }
+    
+    return(tabFrame)
```
[the remark this answers is not in the document]
Comment (gergness) [reply R/tab-book.R:188]: Main takeaway is that I wish the object made more sense on its own, I don't care too much about the messyness inside, but once it leaves the function it should be easier to interpret

File: R/tab-book.R [hunk 2]
```diff
@@ -46,11 +51,149 @@
...
+    }
+
+    # Drop duplicated from last
+    c1 <- !duplicated(
+        tabFrame[!names(tabFrame) %in% "keep"],
+        fromLast = TRUE
+    )
+    tabFrame <- tabFrame[c1,]
+    
+    if (!include_original_weighted) {
```
[the remark this answers is not in the document]
Comment (1beb) [reply R/tab-book.R]: wontfix: leaving it in for downstream include_original_weighted now that it's been removed.

File: R/tab-book.R [hunk 3]
```diff
@@ -46,11 +51,149 @@
...
+        tabFrame <- tabFrame[tabFrame$keep,]
+    } else {
+        tabFrame <- tabFrame[!tabFrame$values %in% unique(customWeights$ind),]
+    }
+    
+    # Add an index value per weight (or non-weight)
+    # This is used in repacking
+    tabFrame$index <- NA
```
[the remark this answers is not in the document]
Comment (1beb) [reply R/tab-book.R:176]: Done

File: R/tab-book.R [hunk 4]
```diff
@@ -46,11 +51,149 @@
...
+    }
+    
+    if (is.list(weight)) {
+        tabFrame <- tabFramePrepare(
+            dataset = dataset, 
+            weight = weight,
+            include_original_weighted = include_original_weighted
+        )
+        
+        books <- list()
```
[the remark this answers is not in the document]
Comment (1beb) [reply R/tab-book.R]: Already done from #510

File: R/tab-book.R [hunk 5]
```diff
@@ -46,11 +51,149 @@
...
+            file = file,
+            filter = filter,
+            use_legacy_endpoint = use_legacy_endpoint,
+            ...
+            )
+        )
+    }
+    
+    if (is.list(weight)) {
+        tabFrame <- tabFramePrepare(
```
[the remark this answers is not in the document]
Comment (1beb) [reply R/tab-book.R:74]: Done!

File: R/tab-book.R [hunk 6]
```diff
@@ -46,11 +51,149 @@
...
+        for (row in seq_len(nrow(tabFrame))) {
+            message(row)
+            tf <- tabFrame[row,]
+            tf$ind[is.na(tf$ind)] <- "UNWEIGHTED"
+            index <- tf$index
+            part <- books[[tf$ind]]@.Data
+            book@.Data[[1]]$analyses[row] <- part[[1]]$analyses[index]
+            book@.Data[[2]][[row]] <- part[[2]][[index]]
+        }
+
```
[the remark this answers is not in the document]
Comment (1beb) [reply R/tab-book.R]: Done!

**Pull request closed without merging**